library/L1_Peripheral/lpc17xx/system_controller.hpp

```diff
@@ -117,273 +53,563 @@ class SystemController final : public sjsu::SystemController
+    struct
+    {
+      /// The 15-bit PLL0 multiplier value ranging from 6 to 512.
+      uint16_t multiplier = 36;
+      /// The 8-bit PLL0 pre-divider value ranging from 1 to 32.
+      uint8_t pre_divider = 1;
+    } pll0 = {};
+
+    /// Configurations for Pll1 (USB PLL).
+    ///
+    /// @see 4.6.9 Procedure for determining PLL1 settings
+    ///      https://www.nxp.com/docs/en/user-guide/UM10360.pdf#page=52
+    struct
+    {
+      /// The PLL1 multiplier select.
+      Pll1::Multiplier multiplier = Pll1::Multiplier::kMultiplyBy4;
+    } pll1 = {};
+
+    /// Configurations for the CPU clock.
+    struct
+    {
+      /// Maximum allowed CPU speed. This value will be 100 MHz or 120 MHz
+      /// depending on the MCU in use.
+      ///
+      /// @see To determine the max CPU speed, see 1.4.1 Part options summary
+      ///      https://www.nxp.com/docs/en/user-guide/UM10360.pdf#page=7
+      units::frequency::hertz_t max_cpu_clock_rate = 100_MHz;
+      /// Clock source for the CPU clock.
+      CpuClockSource clock_source = CpuClockSource::kPll0;
+      /// The 8-bit CPU clock divider. By default, the divider is set to 6 to
+      /// obtain a CPU clock of 48 MHz using the default PLL0 configurations.
+      uint8_t divider = 6;
+    } cpu = {};
+
+    /// Configurations for the USB subsystem clock.
+    struct
+    {
+      /// Clock source for the USB subsystem clock.
+      UsbClockSource clock_source = UsbClockSource::kPll0;
+      /// Clock divider for the USB clock.
+      ///
+      /// @note This value is only used if the PLL0 output is used to drive the
+      ///       USB subsystem.
+      UsbClockDivider divider = UsbClockDivider::kDivideBy6;
+    } usb = {};
+  };
+
+  /// Fixed Internal RC frequency.
+  static constexpr units::frequency::hertz_t kIrcFrequency = 4_MHz;
+
+  /// Fixed RTC frequency.
   static constexpr units::frequency::hertz_t kRTCFrequency = 32'768_Hz;
 
-  /// Pointer to the system controller peripheral in memory.
+  /// The required frequency for the USB clock.
+  static constexpr units::frequency::hertz_t kUsbClockFrequency = 48_MHz;
+
+  /// Pointer reference to the system controller peripheral in memory.
   inline static LPC_SC_TypeDef * system_controller = LPC_SC;
 
-  // TODO(#1140): Migrate to SystemController V2.0
+  /// @param clock_configuration The desired clock configurations for the
+  ///                            system.
+  explicit constexpr SystemController(
+      ClockConfiguration_t & clock_configuration)
+      : clock_configuration_(clock_configuration)
+  {
+  }
+
   void Initialize() override
   {
-    return;
+    units::frequency::hertz_t osc_clk =
+        clock_configuration_.main_oscillator.frequency;
+
+    units::frequency::hertz_t sys_clk = 0_MHz;
+    units::frequency::hertz_t pll_clk = 0_MHz;
+    units::frequency::hertz_t usb_clk = 0_MHz;
+    uint8_t cpu_clock_divider         = clock_configuration_.cpu.divider;
+
```

Comment:
It may be a good idea to set all of the clocks to `IRC` and `sysclk`. Changing the PLL while its connected will cause issues.